ibis/expr/types/relations.py

```diff
@@ -1818,6 +1831,18 @@ def rename(c):
                 c = c.replace("-", "_")
                 return c.lower()
 
+        elif isinstance(substitutions, str):
+
+            def rename(name):
+                return substitutions.format(name=name)
+
+            # Detect the case of missing or extra format string parameters
+            try:
+                invalid = rename("b026fa95") == rename("530505b1")
```

Comment:
Are we trying to prevent a misunderstanding of format string syntax?

Reply:
It's to throw a better error in the case of typos/user error. Otherwise you'll get a column name conflict error, which masks the real error that you just gave a bad format string. A few possible failure modes:

```python
# Typo of `snake_case`
t.relabel("snake_cse")

# Typo of format string
t.relabel("prefix_{name")

# Wrong format parameter
t.relabel("prefix_{column}")

# Using %s formatting instead
t.relabel("prefix_%s")
```

Reply:
can we generate two value and stick them in globals to avoid the magic strings?

Reply:
Would a better comment fix things? It's not clear to me how adding indirection through globals would improve things.

Could instead do something that looks a bit clearer like

```
rename("_unlikely_str1_") == rename("_unlikely_str2_")
```

Reply:
Random looking strings obfuscate intent. Comments help, but why comment when you can name a variable in a way that expresses the intent?

Reply:
I've pushed up a change. Is that sufficient?